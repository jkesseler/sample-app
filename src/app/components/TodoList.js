/* eslint-disable react/no-array-index-key */
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { List, Paper } from '@material-ui/core';

import * as actions from '../state/todos/actions';
import TodoListItem from './TodoListItem';


export const TodoList = ({ todos, toggleTodo, deleteTodo }) => {
  if (todos && todos.length > 0) {
    return (
      <Paper>
        <List style={{ overflow: 'scroll' }}>
          {todos.map((todo, idx) => (
            <TodoListItem
              {...todo}
              key={todo.id}
              divider={idx !== todos.length - 1}
              onButtonClick={() => deleteTodo(todo)}
              onCheckboxToggle={() => toggleTodo(todo)}
            />
          ))}
        </List>
      </Paper>
    );
  }
  return null;
};

TodoList.propTypes = {
  todos: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    text: PropTypes.string.isRequried,
  })),
  toggleTodo: PropTypes.func.isRequired,
  deleteTodo: PropTypes.func.isRequired,
};

TodoList.defaultProps = {
  todos: [],
};

const mapStateToProps = state => ({
  todos: state.todos,
});

const mapDispatchToProps = dispatch => ({
  toggleTodo: todo => dispatch(actions.toggleTodo(todo.id)),
  deleteTodo: todo => dispatch(actions.deleteTodo(todo.id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(TodoList);
