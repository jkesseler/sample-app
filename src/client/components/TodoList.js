/* eslint-disable react/no-array-index-key */
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { List, Paper } from '@material-ui/core';

import * as actions from '../actions/todos';
import TodoListItem from './TodoListItem';


export const TodoList = ({ todos, toggleTodo, removeTodo }) => {
  if (todos && todos.length > 0) {
    return (
      <Paper>
        <List style={{ overflow: 'scroll' }}>
          {todos.map((todo, idx) => (
            <TodoListItem
              {...todo}
              key={todo.id}
              divider={idx !== todos.length - 1}
              onButtonClick={() => removeTodo(todo)}
              onCheckboxToggle={() => toggleTodo(todo)}
            />
          ))}
        </List>
      </Paper>
    );
  }
  return null;
};

TodoList.propTypes = {
  todos: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    text: PropTypes.string.isRequried,
  })),
  toggleTodo: PropTypes.func.isRequired,
  removeTodo: PropTypes.func.isRequired,
};

TodoList.defaultProps = {
  todos: [],
};

const mapStateToProps = state => ({
  todos: state.todos,
});

const mapDispatchToProps = dispatch => ({
  toggleTodo: todo => dispatch(actions.toggleTodo(todo.id)),
  removeTodo: todo => dispatch(actions.removeTodo(todo.id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(TodoList);
