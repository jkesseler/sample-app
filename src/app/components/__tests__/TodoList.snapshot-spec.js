import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import TodoList from '../TodoList';

const defaultProps = {
  todos: [{
    id: '1',
    text: 'some text',
  }],
  toggleTodo: jest.fn(),
  deleteTodo: jest.fn(),
};

describe('TodoList', () => {
  it('renders without crashing', () => {
    const component = shallow(<TodoList />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders with default props', () => {
    const component = shallow(<TodoList {...defaultProps} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
