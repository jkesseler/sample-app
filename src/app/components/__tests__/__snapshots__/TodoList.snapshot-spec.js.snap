// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders with default props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`TodoList renders without crashing 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
